src/js/relay.firefox.com/get_profile_data.js

```diff
@@ -1,419 +1,453 @@
 (async function () {
-  // Get the api token from the account profile page
-  const profileMainElement = document.querySelector("#profile-main");
-  const apiToken = profileMainElement.dataset.apiToken;
-  browser.storage.local.set({ apiToken });
-
-  // API URL is ${RELAY_SITE_ORIGIN}/api/v1/
-  const { relayApiSource } = await browser.storage.local.get("relayApiSource");
-
-  const apiProfileURL = `${relayApiSource}/profiles/`;
-  const apiRelayAddressesURL = `${relayApiSource}/relayaddresses/`;
-
-  async function apiRequest(url, method = "GET", body = null, opts=null) {
-
-    const cookieString =
-      typeof document.cookie === "string" ? document.cookie : "";
-    const cookieStringArray = cookieString
-      .split(";")
-      .map((individualCookieString) => individualCookieString.split("="))
-      .map(([cookieKey, cookieValue]) => [
-        cookieKey.trim(),
-        cookieValue.trim(),
-      ]);
-
-    const [_csrfCookieKey, csrfCookieValue] = cookieStringArray.find(
-      ([cookieKey, _cookieValue]) => cookieKey === "csrftoken"
-    );
+  const dahsboardInitializationObserver = new MutationObserver((mutations) => {
+    mutations.forEach(mutation => {
+      if (mutation.type !== "childList") {
+        return;
+      }
+      mutation.addedNodes.forEach(node => {
+        if (node.id !== "profile-main") {
+          return;
+        }
 
-    browser.storage.local.set({ csrfCookieValue: csrfCookieValue });
+        // Once an element `#profile-main` is added, the dashboard is initialized
+        // with the data needed by the add-on; stop watching for futher changes,
+        // and run once:
+        dahsboardInitializationObserver.disconnect();
+        run();
+      });
+    });
+  });
+  if (document.getElementById("profile-main") === null) {
+    dahsboardInitializationObserver.observe(document.body, { childList: true, subtree: true });
```

Comment:
Yeah we could do that, but that would require us to first do a production deploy of the current website before distributing the updated add-on to the users.

It's also going to be very temporary, as this is currently only used to get the API token ([here](https://github.com/mozilla/fx-private-relay-add-on/blob/e099130eaf5cc8ce66c74c6cb127d008c4ba2b23/src/js/relay.firefox.com/get_profile_data.js#L28-L29)) - which we're moving away from with the new auth mechanism :) Once we've transitioned to that, we can go back to just executing `run()` immediately. So it seems like an acceptable trade-off to me, but happy to discuss.

Reply:
Website prod deploys tend to happen sooner and more often than add-ons already, so I feel fine about that. And I'm willing to be the new auth mechanism will take a while longer than "very" temporary.

Reply:
Hmm OK, so thinking about this some more, I don't even think we'll need an update to the current website, since the MutationObserver is only initialised for the new one. What I could do is update the React front-end to set a `data` attribute on the body the moment that `profile-main` is added to the DOM, so that the MutationObserver only needs to look for changes to that specific attribute. What do you think of that approach?

Reply:
> Yeah we could do that, but that would require us to first do a production deploy of the current website before distributing the updated add-on to the users.
> 
> It's also going to be very temporary, as this is currently only used to get the API token ([here](https://github.com/mozilla/fx-private-relay-add-on/blob/e099130eaf5cc8ce66c74c6cb127d008c4ba2b23/src/js/relay.firefox.com/get_profile_data.js#L28-L29)) - which we're moving away from with the new auth mechanism :) Once we've transitioned to that, we can go back to just executing `run()` immediately. So it seems like an acceptable trade-off to me, but happy to discuss.

This may no longer apply but we definitely can land a small change to the main website first. Is this still applicable?

Reply:
Nope!

> I don't even think we'll need an update to the current website

So the question is: do we want to explicitly add an attribute to the `<body>` at the same time we add `#profile-main` to the DOM? It's slightly more error-prone (in case "at the same time" happens to not be at the same time), but should be more performant.

Reply:
So the failure states are degraded performance vs. broken functionality? Let's play it safe first - i.e., add the mutation observers to `document.body` for now. If it causes significant performance problems in production, we know we can try this technique of adding the `data` attribute to `profile-main`.